DeratControl.Application/Perimeters/Commands/AddPerimeterCommand.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using DeratControl.Application.Interfaces;
+using DeratControl.Application.Requests;
+using DeratControl.Application.Requests.Interfaces;
+using DeratControl.Domain.Entities;
+using DeratControl.Domain.Root.Repositories;
+using DeratControl.Domain.Root.Exceptions;
+namespace DeratControl.Application.Perimeters.Commands
+{   /// <summary>
+/// DTO class for adding perimeters command
+/// </summary>
+    public class AddPerimeterCommand : IRequest
+    {
+        public int FacilityId { get; set; }
+        public virtual Facility Facility { get; set; }
+        public PerimeterType PerimeterType { get; set; }
+    }
+    /// <summary>
+    /// class that will implement CommandHandler : Add Perimeter to facility
+    /// </summary>
+    public class AddPerimeterCommandHandler : BaseCommandHandler<AddPerimeterCommand>
+    {
+        private readonly IFacilityRepository _repository;
+        //TODO : inject IUnitOfWork instead of IFacilityRepository when available
+        public AddPerimeterCommandHandler(IFacilityRepository repository)
+        {
+            this._repository = repository;
+        }
+        protected override async Task<CommandResult> HandleRequest(CommandExecutionContext executionContext, AddPerimeterCommand request)
```

Comment:
I'd recommend the next steps

1) Get Facility object with perimeters, if the facility doesn't exist throw an exception
2) change method AddPerimeter to accept 2 arguments: perimeter type and user
3) inside the method AddPerimeter check if the perimeter with the provided type already exists in a Perimeters collection, if exists - throw an exception, otherwise create a new object of Perimeter and add it to collection
4) use unitofwork to save changes

Reply:
done